### ISSUE TITLE
"Error c2280" occurred when generating using vs2017 15.9.45

### ISSUE THREAD

**Issue by iwetuan** [djurodrljaca/CppConfigFramework#4]
I have compiled cedarframework by using vs2017 x64, 
but there is an error in compiling cppconfigframework at 
configreaderregistry line 98 
STD:: Map \< qstring, STD:: unique_ ptr\<ConfigReaderBase\>\> m_ configReaders;
 How can I solve this problem and compile it

**Comment by iwetuan** [comment]
The change to this form was passed in vs2017 x64
“”std::map\<QString, std::shared_ ptr\<ConfigReaderBase\>\> m_ configReaders;“”

**Comment by iwetuan** [comment]
This can also solve the problem, but I don't know the reason

configReaderRegistry.cpp
`
ConfigReaderRegistry::ConfigReaderRegistry():
m_configReaders(std::make_unique<std::map<QString, std::unique_ptr<ConfigReaderBase>>>())
{
    registerConfigReader(QStringLiteral("CppConfigFramework"), std::make_unique<ConfigReader>());
}
`

//ConfigReaderRegistry.hpp
`
std::unique_ptr<std::map<QString, std::unique_ptr<ConfigReaderBase>>> m_configReaders;`

**Comment by iwetuan** [comment]
according to https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/compiler-error-c2280?view=msvc-170

another solution as this
class CPPCONFIGFRAMEWORK_EXPORT ConfigReaderRegistry
{
public:
    //! Destructor
    ~ConfigReaderRegistry() = default;

   // //! Copy constructor is disabled
    **ConfigReaderRegistry(const ConfigReaderRegistry &) = delete;**
	**ConfigReaderRegistry &operator=(const ConfigReaderRegistry &) = delete;**

**Comment by djurodrljaca** [comment]
Yes, the last one seems to be the best solution. Please check if [this commit](https://github.com/djurodrljaca/CppConfigFramework/commit/a013feb1beb92c61901496263612983c7ec19b83) fixes the issue.

**Comment by iwetuan** [comment]
yes it works
